mpi-proxy-split/mpi-wrappers/mpi_collective_p2p.c

```diff
@@ -485,39 +485,58 @@ int MPI_Reduce_scatter(const void* sendbuf, void* recvbuf,
 int MPI_Scan(const void* sendbuf, void* recvbuf, int count,
              MPI_Datatype datatype, MPI_Op op, MPI_Comm comm) {
   PROLOG_Comm_rank_size;
+
+  // Get MPI datatype size to allocate buffers
   int root = 0;
   MPI_Aint lower_bound;
   MPI_Aint extent;
   MPI_Type_get_extent(datatype, &lower_bound, &extent);
+  int retval;
+
+  // Deal with buffers in place
```

Comment:
Please change the line below to also include `FORTRAN_MPI_IN_PLACE`:
```
int inplace = (sendbuf == MPI_IN_PLACE || sendbuf == FORTRAN_MPI_IN_PLACE );
```
And please make the same change everywhere where you see `MPI_IN_PLACE`. 

This should probably be done as a separate commit in this PR.  This is a bug in this code.